### PR TITLE
🚀 Github Actions Workflows

### PR DESCRIPTION
**Pull request by jakoch**
This PR adds the following workflows: `Build On Windows`, `Build Documentation` and `CodeQL Analysis`.

- The `Build On Windows` workflow is a build matrix for Windows with triplet and PCH compile flag support.
   - Build Artifacts are uploaded per matrix run.
   - Test Artifacts are uploaded into a shared test artifact and then published by the job `publish-test-results` after the matrix finishes, so that tests statistics of all runs are aggregated and displayed.
- The `Build Documentation` workflow triggers the (re-)building of the documentation on the docs repo.
   - It uses the GithubToken to make the `workflow_dispatch` request.
   - I couldn't trigger/test this from a fork. I hope the permission level fits, when the request is made from the official repo.
- The `CodeQL Analyze` workflow builds the x64-windows-static triplet with CodeQL scanning enabled and uploads the scan report.
   - The config resides in `./.github/codeql/codeql-config.yml` and is using the `security-and-quality` scan.
   - Other scans are prepared, but commented out. The Config needs tweaking. 
   - Reports should appear under the "Security" tab.
- The `Build On Windows` and `CodeQL Analyze` depend on the job `install-vulkan-sdk-and-runtime`.
    - This is an independend job, because the cache is shared by all matrix runs.
    - The job resides in both files, because it's not supported on GHA (yet) to call jobs from other workflow files.
- `Build On Windows` depends on an external file `.vsconfig`, which contains the description of required Visual Studio Components. The file was exported from VSInstaller.

❗   **Expected failure** on the PCH:OFF run. A pull request containing the header fixes will follow.

Known Issues of Github Actions i ran into: 
- Status badges for each run of the build matrix are not supported, yet 
- Running a dependend job from another workflow file doesn't work, yet
- Build artifact is zipped twice. Not fixed, yet.
- 🙄 

Please review and comment on the code, so that we can get it into a state, where it is ready to be merged.

### CODE REVIEW

File: .github/workflows/build-on-windows.yml [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    name: ${{ matrix.config.VCPKG_TARGET_TRIPLET }} PCH:${{ matrix.config.TT_BUILD_PCH }}
+    needs: install-vulkan-sdk-and-runtime
+    # https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md
+    runs-on: windows-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - { VCPKG_TARGET_TRIPLET: x64-windows-static, BUILD_SHARED_LIBS: OFF, TT_BUILD_PCH: ON}
```
Comment (takev) [inline .github/workflows/build-on-windows.yml:30]: I changed ttauri so that BUILD_SHARED_LIBS is now automatically gets set by the VCPKG_TARGET_TRIPLET.

Comment (takev) [inline .github/workflows/build-on-windows.yml:30]: I wonder for performance reasons if we should only build with TT_BUILD_PCH: OFF.
That way we do a proper source code test.
At least the OFF test should be done first, so we can only make that one "required" when merging.

File: .github/workflows/build-on-windows.yml [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+      - name: ❔ CHECK important folders, to see if everything is present (before building)
+        run: |
+          dir %VULKAN_SDK%\Bin
+          dir %VULKAN_SDK%\runtime\%PLATFORM%
+          dir %LocalAppData%\vcpkg
+
+      - name: ✏️ CMake ➔ Configure (including VCPKG ➔ Install Dependencies)
+        run: |
+          cmake -G "Visual Studio 16 2019" -A ${{ env.PLATFORM }}                           ^
```
Comment (takev) [inline .github/workflows/build-on-windows.yml:109]: As with appveyor it seems artifacts must be relative to the source directory.
It is probably best (which is also what most install manuals would say when building with cmake) to create
a build directory inside the source directory. cd into the build and then do the same without the -B option and with .. as the last argument.

That way you can directly execute build and install from that same directory as well.

File: .github/workflows/build-on-windows.yml [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                -DTT_BUILD_PCH=${{ matrix.config.TT_BUILD_PCH }}                            ^
+                -DCMAKE_VERBOSE_MAKEFILE=ON
+
+      - name: 🙏 CMake ➔ Build
+        run: |
+          cd ..\build
+          cmake --build . --config %BUILD_TYPE% -j %NUMBER_OF_PROCESSORS%
+
+      - name: 📦 CMake ➔ Install
+        run: cmake --install ..\build --config %BUILD_TYPE% --prefix ..\install --strip --verbose
```
Comment (takev) [inline .github/workflows/build-on-windows.yml:124]: Install into a subdirectory of the build directory. Already using a variable to the final name of the artifact before zipping. That way you do not have to rename the directory later.

File: .github/workflows/build-on-windows.yml [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+      # Build Artifact Name: ttauri-0.3.0-0cda6a2-MSVC19-x64-static-RelWithDebInfo
+      - name: ✏️ Fetch Versioning Data & Set Artifact Name
+        shell: pwsh
+        run: |
+          $NAME=$(jq -r .name vcpkg.json)
+          $VERSION=$(jq -r .version vcpkg.json)
+          $SHORT_HASH=$($env:GITHUB_SHA.substring(0,7))
+          $COMPILER="MSVC19"
+          $TRIPLET=$("${{ matrix.config.VCPKG_TARGET_TRIPLET }}".Replace("-windows",""))
```
Comment (takev) [inline .github/workflows/build-on-windows.yml:166]: I rather keep -windows as part of the artifact especially when we get -linux versions.

**Review by takev — CHANGES_REQUESTED** [review]

**Comment by jakoch** [comment]
I wonder why the first push of this PR, went green on the tests for the PCH:ON build.
The logger [prints during tests](https://github.com/ttauri-project/ttauri/runs/2181808601#step:14:209), but the test run doesn't crash, it actually [passes](https://github.com/ttauri-project/ttauri/runs/2181808601#step:14:788).  

The changes regarding the folder structure? Executables are build, resources for tests are there.
Hmm.. maybe globbing fails due to `\version-0.3.0\`?

The only other change is  `Release` to `RelWithDebInfo`? Does this have an influence on the log level?

**Review by takev — APPROVED** [review]

**Pull request merged**